cmd/strelaysrv/main.go

```diff
@@ -306,15 +327,18 @@ func main() {
 }
 
 func monitorLimits() {
-	limitCheckTimer = time.NewTimer(time.Minute)
+	limitCheckTimer = time.NewTimer(limitCheckInterval)
 	for range limitCheckTimer.C {
 		if numConnections.Load()+numProxies.Load() > descriptorLimit {
-			overLimit.Store(true)
-			log.Println("Gone past our connection limits. Starting to refuse new/drop idle connections.")
-		} else if overLimit.CompareAndSwap(true, false) {
-			log.Println("Dropped below our connection limits. Accepting new connections.")
+			if !overLimit.Swap(true) {
+				log.Println("Gone past our connection limits. Starting to refuse new/drop idle connections.")
+			}
+		} else {
+			if overLimit.CompareAndSwap(true, false) {
+				log.Println("Dropped below our connection limits. Accepting new connections.")
+			}
+			limitCheckTimer.Reset(limitCheckInterval)
```

Comment:
Why is this moving into the else clause? 
Given this is a timer, not a ticker, C fires only once (or once after reset)